src/NHibernate.Test/Legacy/ParentChildTest.cs

```diff
@@ -372,10 +372,15 @@ public void CollectionQuery()
 			l.Add(null);
 			l.Add(s2);
 			c.ManyToMany = l;
+			c.One = new One { Value = "one" };
+			s.Save(c.One);
 			s.Save(c);
 
 			Assert.AreEqual(1,
 			                s.CreateQuery("select c from c in class ContainerX, s in class Simple where c.OneToMany[2] = s").List
+			                	().Count);			
+			Assert.AreEqual(1,
+			                s.CreateQuery("select c from c in class ContainerX, s in class Simple where c.OneToMany[2] = s and c.One != null").List
```

Comment:
I'm a bit lost.. This check works on my machine with both Sql Server and SQLite.. How come it fails on CI???

Reply:
Do you run .net 6 tests? Or .net 4.6.1?

Reply:
net6.0

Reply:
I'm still puzzled why original tests fails on linux (seems it's the only difference). But fail is not related to PR changes (as this check is about proper SQL generation with valid order of joins)